src/bootstrap/native.rs

```diff
@@ -1029,7 +1036,14 @@ impl Step for Sanitizers {
         // Unfortunately sccache currently lacks support to build them successfully.
         // Disable compiler launcher on Darwin targets to avoid potential issues.
         let use_compiler_launcher = !self.target.contains("apple-darwin");
-        configure_cmake(builder, self.target, &mut cfg, use_compiler_launcher, LdFlags::default());
+        configure_cmake(
+            builder,
+            self.target,
+            &mut cfg,
+            use_compiler_launcher,
+            LdFlags::default(),
+            &["-fembed-bitcode=off"],
```

Comment:
This is required because sanitizers are built with the `-U` linker flag, which isn't compatible with bitcode. The sanitizers shipped in Xcode also don't have bitcode, and theoretically that shouldn't ever matter since on iOS folks don't ship sanitizers to prod. This wouldn't be necessary if bitcode was dropped entirely now that Xcode 14 deprecated it and Apple no longer accepts apps with it, see https://github.com/rust-lang/cc-rs/pull/768